Docker: Chat tab broken — npm install fails with EACCES due to root-owned node_modules

## Summary

When running the dashboard via Docker with `HERMES_DASHBOARD_TUI=1`, the Chat tab immediately shows `[session ended]` and no input is accepted.

## Root Cause

`/opt/hermes/ui-tui` and its subdirectories (including `packages/hermes-ink/node_modules`) are owned by `root` in the Docker image, but the dashboard process runs as uid 10000 (`hermes`) after privilege drop in `entrypoint.sh`.

When the Chat tab opens a WebSocket connection, `_make_tui_argv()` in `hermes_cli/main.py` calls `npm install` in `/opt/hermes/ui-tui`. This fails with:

```
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/opt/hermes/ui-tui/packages/hermes-ink/node_modules/@esbuild/aix-ppc64'
```

`_make_tui_argv` then calls `sys.exit(1)`, which is caught as `SystemExit` in `pty_ws()` (web_server.py line ~3007) and results in the WebSocket closing immediately — shown to the user as `[session ended]`.

The same issue affects `/opt/hermes/node_modules`.

## Steps to Reproduce

1. Run `hermes-dashboard` via Docker Compose (standard setup from docs)
2. Set `HERMES_DASHBOARD_TUI=1`
3. Open the dashboard and click the **Chat** tab
4. Observe `[session ended]` — typing is not possible

## Fix

Add a `chown` in the Dockerfile before the image is finalized:

```dockerfile
RUN chown -R hermes:hermes /opt/hermes/ui-tui /opt/hermes/node_modules
```

This ensures the `hermes` user (uid 10000) can write to `node_modules` when `npm install` runs at Chat tab startup.

## Environment

- Image: `nousresearch/hermes-agent:latest` (v0.12.0, 2026.4.30)
- `HERMES_DASHBOARD_TUI=1`
- Linux/amd64, Docker Compose

--- a/Dockerfile
+++ b/Dockerfile
@@ -66,8 +66,14 @@ RUN cd web && npm run build && \
 # ---------- Permissions ----------
 # Make install dir world-readable so any HERMES_UID can read it at runtime.
 # The venv needs to be traversable too.
+# node_modules trees additionally need to be writable by the hermes user
+# so the runtime `npm install` triggered by _tui_need_npm_install() in
+# hermes_cli/main.py succeeds (see #18800). /opt/hermes/web is build-time
+# only (HERMES_WEB_DIST points at hermes_cli/web_dist) and is intentionally
+# not chowned here.
 USER root
-RUN chmod -R a+rX /opt/hermes
+RUN chmod -R a+rX /opt/hermes && \
+    chown -R hermes:hermes /opt/hermes/ui-tui /opt/hermes/node_modules
 # Start as root so the entrypoint can usermod/groupmod + gosu.
 # If HERMES_UID is unset, the entrypoint drops to the default hermes user (10000).
 

--- a/tests/tools/test_dockerfile_node_modules_perms.py
+++ b/tests/tools/test_dockerfile_node_modules_perms.py
@@ -1,0 +1,39 @@
+"""contract test: dockerfile chowns runtime node_modules trees to hermes
+
+regression guard for #18800. the container drops privileges to the hermes
+user (uid 10000) in entrypoint.sh, then the TUI launcher's
+_tui_need_npm_install() trips on every startup (see the
+npm_config_install_links=false comment in the Dockerfile) and runs
+`npm install` in /opt/hermes/ui-tui. that install fails with EACCES unless
+the runtime node_modules trees are owned by hermes.
+"""
+from __future__ import annotations
+
+from pathlib import Path
+
+REPO_ROOT = Path(__file__).resolve().parents[2]
+DOCKERFILE = REPO_ROOT / "Dockerfile"
+
+
+def test_dockerfile_chowns_runtime_node_modules_to_hermes_user() -> None:
+    text = DOCKERFILE.read_text()
+
+    chown_lines = [
+        line for line in text.splitlines()
+        if "chown" in line and "hermes:hermes" in line
+    ]
+    assert chown_lines, (
+        "Dockerfile must contain a chown -R hermes:hermes for the runtime "
+        "node_modules trees; see #18800"
+    )
+
+    chown_block = "\n".join(chown_lines)
+
+    # both runtime-mutable trees must be passed to the chown command.
+    # /opt/hermes/web is intentionally excluded: it is build-time only,
+    # because HERMES_WEB_DIST points at hermes_cli/web_dist for runtime.
+    for required_path in ("/opt/hermes/ui-tui", "/opt/hermes/node_modules"):
+        assert required_path in chown_block, (
+            f"{required_path} must be passed to a chown -R hermes:hermes "
+            f"command in the Dockerfile (see #18800)"
+        )

